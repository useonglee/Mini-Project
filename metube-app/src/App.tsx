import { Routes, Route, Link } from "react-router-dom";

// Auto generates routes from files under ./pages
// https://vitejs.dev/guide/features.html#glob-import
const pages = import.meta.globEager("./pages/*.tsx");

const routes = Object.keys(pages).map((path: string) => {
  const name = path.match(/\.\/pages\/(.*)\.tsx$/)[1];

  return {
    name,
    path: name === "Home" ? "/" : `/${name.toLowerCase()}`,
    component: pages[path].default,
  };
});

export default function App() {
  return (
    <>
      <nav>
        <ul>
          {routes.map(({ name, path }) => {
            return (
              <li key={path}>
                <Link to={path}>{name}</Link>
              </li>
            );
          })}
        </ul>
      </nav>
      <Routes>
        {routes.map(({ name, path, component: RouteComp }) => {
          return <Route key={name} path={path} element={<RouteComp />} />;
        })}
      </Routes>
    </>
  );
}
