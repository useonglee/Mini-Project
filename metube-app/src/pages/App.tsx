import { Route, Routes } from "react-router-dom";
import DetailPage from "./detail";
import HomePage from "./Home";

function App() {
  return (
    <Routes>
      <Route path="/" element={<HomePage />} />
      <Route path="/watch" element={<DetailPage />} />
    </Routes>
  );
}

export default App;
