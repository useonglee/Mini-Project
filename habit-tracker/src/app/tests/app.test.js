import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import HabitPresenter from "../habit_presenter";
import App from "../app";

describe("App CI test", () => {
  let habitPresenter;

  beforeAll(() => {
    habitPresenter = new HabitPresenter([
      { id: 1, name: "Reading", count: 0 },
      { id: 2, name: "Running", count: 0 },
      { id: 3, name: "Coding", count: 0 },
    ]);
  });

  it("render App component", () => {
    const component = renderer.create(<App presenter={habitPresenter} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it("counts test", () => {
    render(<App presenter={habitPresenter} />);
    const button = screen.getAllByTitle("increase")[0];
    const count = screen.getByTestId("total-count");

    userEvent.click(button);

    expect(count.innerHTML).toBe("1");
  });

  it("Add new habit", () => {
    render(<App presenter={habitPresenter} />);

    const input = screen.getByPlaceholderText("Habit");
    const button = screen.getByText("Add");
    userEvent.type(input, "New Habit");
    userEvent.click(button);

    const addName = screen.getAllByTestId("habit-name")[3];
    expect(addName.innerHTML).toBe("New Habit");

    const addedCount = screen.getAllByTestId("habit-count")[3];
    expect(addedCount.innerHTML).toBe("0");
  });

  it("resets all counters", () => {
    render(<App presenter={habitPresenter} />);
    const button = screen.getByText("Reset All");

    userEvent.click(button);

    screen.getAllByTestId("habit-count").forEach((count) => {
      expect(count.innerHTML).toBe("0");
    });
  });
});
