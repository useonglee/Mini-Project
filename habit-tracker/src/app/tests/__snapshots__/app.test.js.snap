// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App CI test render App component 1`] = `
Array [
  <nav
    className="navbar"
  >
    <i
      className="navbar-logo fas fa-leaf"
    />
    <span>
      Habit Tracker
    </span>
    <span
      className="navbar-count"
      data-testid="total-count"
    >
      0
    </span>
  </nav>,
  <div
    className="habits"
  >
    <form
      className="add-form"
      onSubmit={[Function]}
    >
      <input
        className="add-input"
        placeholder="Habit"
        type="text"
      />
      <button
        className="add-button"
      >
        Add
      </button>
    </form>
    <ul>
      <li
        className="habit"
      >
        <span
          className="habit-name"
          data-testid="habit-name"
        >
          Reading
        </span>
        <span
          className="habit-count"
          data-testid="habit-count"
        >
          0
        </span>
        <button
          className="habit-button habit-increase"
          onClick={[Function]}
          title="increase"
        >
          <i
            className="fas fa-plus-square"
          />
        </button>
        <button
          className="habit-button habit-decrease"
          onClick={[Function]}
          title="decrease"
        >
          <i
            className="fas fa-minus-square"
          />
        </button>
        <button
          className="habit-button habit-delete"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash"
          />
        </button>
      </li>
      <li
        className="habit"
      >
        <span
          className="habit-name"
          data-testid="habit-name"
        >
          Running
        </span>
        <span
          className="habit-count"
          data-testid="habit-count"
        >
          0
        </span>
        <button
          className="habit-button habit-increase"
          onClick={[Function]}
          title="increase"
        >
          <i
            className="fas fa-plus-square"
          />
        </button>
        <button
          className="habit-button habit-decrease"
          onClick={[Function]}
          title="decrease"
        >
          <i
            className="fas fa-minus-square"
          />
        </button>
        <button
          className="habit-button habit-delete"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash"
          />
        </button>
      </li>
      <li
        className="habit"
      >
        <span
          className="habit-name"
          data-testid="habit-name"
        >
          Coding
        </span>
        <span
          className="habit-count"
          data-testid="habit-count"
        >
          0
        </span>
        <button
          className="habit-button habit-increase"
          onClick={[Function]}
          title="increase"
        >
          <i
            className="fas fa-plus-square"
          />
        </button>
        <button
          className="habit-button habit-decrease"
          onClick={[Function]}
          title="decrease"
        >
          <i
            className="fas fa-minus-square"
          />
        </button>
        <button
          className="habit-button habit-delete"
          onClick={[Function]}
        >
          <i
            className="fas fa-trash"
          />
        </button>
      </li>
    </ul>
    <button
      className="habits-reset"
      onClick={[Function]}
    >
      Reset All
    </button>
  </div>,
]
`;
